src/providers/java_maven.js

```diff
@@ -216,6 +196,60 @@ export default class Java_maven extends Base_java {
 		return rootDependency
 	}
 
+	#selectMvnRuntime(manifestPath, opts) {
```

Comment:
Do you think this would work for gradlew? If so, can we make it generic or you prefer to add it in a follow up PR?

Reply:
Im reasonably confident this can be made generic. I was originally going to do gradle in a follow up PR

Reply:
Perfect.